src/nccl_ofi_rdma.c

```diff
@@ -3013,6 +3013,12 @@ static int recv(nccl_net_ofi_recv_comm_t *recv_comm, int n, void **buffers,
 
 	assert(r_comm != NULL);
 
+	if (r_comm->comm_active == false) {
```

Comment:
I'm not sure i understand why we have this marker.  It's set to false right before `recv_comm_destroy()`, at which point this memory is out of scope.  So this feels like a useless check.

Reply:
> I'm not sure i understand why we have this marker. It's set to false right before `recv_comm_destroy()`, at which point this memory is out of scope. So this feels like a useless check.

The deferred cleanup feature, introduced in a later commit in this series, avoids calling `{recv/send}_comm_destroy` until close handshaking is complete. So this is just a sanity check to prevent NCCL from trying to use a communicator after it closes it and before the resources are actually destroyed.